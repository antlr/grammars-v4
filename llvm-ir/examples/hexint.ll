define void @f() {
	add i4 s0x0, 0
	add i4 s0x1, 1
	add i4 s0x2, 2
	add i4 s0x3, 3
	add i4 s0x4, 4
	add i4 s0x5, 5
	add i4 s0x6, 6
	add i4 s0x7, 7
	add i4 s0x8, -8
	add i4 s0x9, -7
	add i4 s0xA, -6
	add i4 s0xB, -5
	add i4 s0xC, -4
	add i4 s0xD, -3
	add i4 s0xE, -2
	add i4 s0xF, -1
	ret void
}

define void @g() {
	add i4 u0x0, 0
	add i4 u0x1, 1
	add i4 u0x2, 2
	add i4 u0x3, 3
	add i4 u0x4, 4
	add i4 u0x5, 5
	add i4 u0x6, 6
	add i4 u0x7, 7
	add i4 u0x8, 8
	add i4 u0x9, 9
	add i4 u0xA, 10
	add i4 u0xB, 11
	add i4 u0xC, 12
	add i4 u0xD, 13
	add i4 u0xE, 14
	add i4 u0xF, 15
	ret void
}
