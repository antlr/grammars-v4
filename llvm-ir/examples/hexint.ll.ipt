compilationUnit
 topLevelEntity
  funcDef
   define
   funcHeader
    type
     void
    @f
    (
    params
    )
   funcBody
    {
    basicBlock
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0x0
        ,
        value
         constant
          intConst
           0
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0x1
        ,
        value
         constant
          intConst
           1
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0x2
        ,
        value
         constant
          intConst
           2
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0x3
        ,
        value
         constant
          intConst
           3
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0x4
        ,
        value
         constant
          intConst
           4
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0x5
        ,
        value
         constant
          intConst
           5
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0x6
        ,
        value
         constant
          intConst
           6
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0x7
        ,
        value
         constant
          intConst
           7
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0x8
        ,
        value
         constant
          intConst
           -8
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0x9
        ,
        value
         constant
          intConst
           -7
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0xA
        ,
        value
         constant
          intConst
           -6
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0xB
        ,
        value
         constant
          intConst
           -5
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0xC
        ,
        value
         constant
          intConst
           -4
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0xD
        ,
        value
         constant
          intConst
           -3
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0xE
        ,
        value
         constant
          intConst
           -2
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            s0xF
        ,
        value
         constant
          intConst
           -1
     terminator
      retTerm
       ret
       void
    }
 topLevelEntity
  funcDef
   define
   funcHeader
    type
     void
    @g
    (
    params
    )
   funcBody
    {
    basicBlock
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0x0
        ,
        value
         constant
          intConst
           0
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0x1
        ,
        value
         constant
          intConst
           1
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0x2
        ,
        value
         constant
          intConst
           2
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0x3
        ,
        value
         constant
          intConst
           3
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0x4
        ,
        value
         constant
          intConst
           4
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0x5
        ,
        value
         constant
          intConst
           5
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0x6
        ,
        value
         constant
          intConst
           6
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0x7
        ,
        value
         constant
          intConst
           7
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0x8
        ,
        value
         constant
          intConst
           8
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0x9
        ,
        value
         constant
          intConst
           9
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0xA
        ,
        value
         constant
          intConst
           10
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0xB
        ,
        value
         constant
          intConst
           11
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0xC
        ,
        value
         constant
          intConst
           12
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0xD
        ,
        value
         constant
          intConst
           13
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0xE
        ,
        value
         constant
          intConst
           14
     instruction
      valueInstruction
       addInst
        add
        typeValue
         firstClassType
          concreteType
           intType
            i4
         value
          constant
           intConst
            u0xF
        ,
        value
         constant
          intConst
           15
     terminator
      retTerm
       ret
       void
    }
 <EOF>
