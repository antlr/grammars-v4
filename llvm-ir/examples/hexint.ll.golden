define void @f() {
0:
	%1 = add i4 0, 0
	%2 = add i4 1, 1
	%3 = add i4 2, 2
	%4 = add i4 3, 3
	%5 = add i4 4, 4
	%6 = add i4 5, 5
	%7 = add i4 6, 6
	%8 = add i4 7, 7
	%9 = add i4 -8, -8
	%10 = add i4 -7, -7
	%11 = add i4 -6, -6
	%12 = add i4 -5, -5
	%13 = add i4 -4, -4
	%14 = add i4 -3, -3
	%15 = add i4 -2, -2
	%16 = add i4 -1, -1
	ret void
}

define void @g() {
0:
	%1 = add i4 0, 0
	%2 = add i4 1, 1
	%3 = add i4 2, 2
	%4 = add i4 3, 3
	%5 = add i4 4, 4
	%6 = add i4 5, 5
	%7 = add i4 6, 6
	%8 = add i4 7, 7
	%9 = add i4 8, 8
	%10 = add i4 9, 9
	%11 = add i4 10, 10
	%12 = add i4 11, 11
	%13 = add i4 12, 12
	%14 = add i4 13, 13
	%15 = add i4 14, 14
	%16 = add i4 15, 15
	ret void
}
